core/src/data.ts

```diff
@@ -44,50 +44,65 @@ const DEFAULT_DRUM_PITCH_CLASSES: number[][] = [
   [51, 52, 53, 59, 82]
 ];
 
+export interface MelodyConverterSpec {
+  type: 'MelodyConverter';
+  args: MelodyConverterArgs;
+}
+export interface DrumsConverterSpec {
+  type: 'DrumsConverter';
+  args: DrumsConverterArgs;
+}
+export interface DrumRollConverterSpec {
+  type: 'DrumRollConverter';
+  args: DrumsConverterArgs;
+}
+export interface TrioConverterSpec {
+  type: 'TrioConverter';
+  args: TrioConverterArgs;
+}
+export interface DrumsOneHotConverterSpec {
+  type: 'DrumsOneHotConverter';
+  args: DrumsConverterArgs;
+}
+
 /**
```

Comment:
Doc is no longer true, but I think it should be. Can we keep an interface that all of the specs extend so that they have to have a string type and a BaseConverterArgs?

Reply:
If I set it up that way, the converterFromSpec function doesn't type check and I have to cast all of the different arg types.  I'm pretty sure the way I did it is idiomatic TypeScript.

Reply:
It's what they describe here if you search for "discriminated union": http://www.typescriptlang.org/docs/handbook/advanced-types.html

Reply:
Okay. Let's move this docstring and the expert type above the specs.

Reply:
Just to verify, there is no way to make sure these args are always BaseConverterArgs?

Reply:
The super(args) call wouldn't typecheck if they weren't.